### PR TITLE
CSOT: Added timeout to settings & connection string

### PR DESCRIPTION
**Pull request by rozza**
JAVA-4064

### CODE REVIEW

**Comment by rozza** [comment]
Note: I tried to remain consistent with the rest of the code.  So added the `MS` suffix where appropriate. Also in the legacy driver the API for timeouts returns milliseconds and doesn't take a `TimeUnit`.

File: driver-core/src/main/com/mongodb/ConnectionString.java [hunk 1]
```diff
@@ -280,6 +282,7 @@ public class ConnectionString {
     private Integer maxConnectionLifeTime;
     private Integer maxConnecting;
     private Integer connectTimeout;
+    private Long timeout;
```
Comment (katcharov) [inline driver-core/src/main/com/mongodb/ConnectionString.java:285]: should this have the unit, `timeoutMS`?

**Review by katcharov — APPROVED** [review]
LGTM, and a question

File: driver-core/src/main/com/mongodb/ConnectionString.java (hunk 1 above)
Comment (rozza) [reply driver-core/src/main/com/mongodb/ConnectionString.java:285]: Added to the javadocs its milliseconds. Its annoying that its inconsistent but I opted to keep it consistently inconsistent as not all timeouts will be deprecated.

**Pull request merged**